lib/spack/spack/test/database.py

```diff
@@ -436,3 +436,16 @@ def test_external_entries_in_db(database):
     assert rec.spec.external_path == '/path/to/external_tool'
     assert rec.spec.external_module is None
     assert rec.explicit is True
+
+
+@pytest.mark.regression('8036')
+def test_regression_issue_8036(mutable_database):
```

Comment:
I wish this test didn't depend on the existence of a system path - would it be possible to just set `Spec.prefix` to some path you create with a pytest fixture?

Reply:
Seems legit. I patched `os.path.isdir` for this particular regression test.